clarity/src/vm/types/serialization.rs

```diff
@@ -452,275 +544,444 @@ impl Value {
                 }
             };
 
-            assert!(
-                expect_size as u64 >= bytes_read,
-                "Deserialized more bytes than expected size during deserialization. Expected size = {}, bytes read = {}, type = {}",
-                expect_size,
-                bytes_read,
-                expected_type,
-            );
+            if expect_size as u64 > bytes_read {
```

Comment:
Before it was an assertion, so I think here it should be bytes_read  > expect_size as u64